builder/azure/common/vault.go

```diff
@@ -38,18 +43,17 @@ type SecretAttributes struct {
 
 func (client *VaultClient) GetSecret(vaultName, secretName string) (*Secret, error) {
 	p := map[string]interface{}{
-		"secret-name": secretName,
+		"secret-name": autorest.Encode("path", secretName),
```

Comment:
IMHO, it's strange that autorest.WithPathParameters (called below) doesn't encode the parameters. Should we fix this in autorest?